### ISSUE TITLE
Operator should remove artifacts on `APISource` CR deletion

### ISSUE THREAD

**Issue by lbroudoux** [microcks/microcks-operator#52]
### Reason/Context

As of today, removing an `APISource` CR only deletes the previously created `importers` but not the installed `artifacts`. 

This is because there's a direct identity translation of the `importer` CR concept into a Microcks concept and because Microcks importer's id is stored within the CR status.

On the other hand, there's no direct translation between an `artifact` in the CR and a concept in Microcks. Installing an artifact in a Microcks instance may bring a new API or may enrich an existing API.

However, we realized that this difference in behavior is confusing for users who expect that dropping an `APISource` CR would drop everything into the associated Microcks instance. 

I would say that the current behavior is actually the correct one regarding Microcks semantics, but it definitely does not fit the most with GitOps provisioning use cases of a Microcks instance.

### Description

I propose switching to a default mode where deleting an `APISource` CR referencing `artifacts` actually deletes the API that was created or updated with those artifacts. For that, we need to solve [this issue](https://github.com/microcks/microcks/issues/1398) first.

I also propose introducing a switch like `keepAPIOnDelete: true` on CR to allow bypassing this cleaning and sticking to the semantically correct approach that was the previous behavior. What do you think?

### Implementation ideas

On the technical side of things, the API that pushes artifact into Microcks does not return the technical id of the API | Service contributed but rather the "functional" identifier (service name + service version). It would be easier to [generalize this other enhancement first](https://github.com/microcks/microcks/issues/1225) to avoid additional API roundtrips too.

**Comment by lbroudoux** [comment]
Must now be available in the latest container image produced by CI. This should work well when applied to a `nightly` version of Microcks instance. @mmatur Would you be able to validate this?

**Comment by mmatur** [comment]
> nightly

Thanks @lbroudoux, I tried but I'm still not able to delete an APISource resource. 

<img width="1079" alt="image" src="https://github.com/user-attachments/assets/90950e51-7dc2-431e-91e2-51fb78257d65">

**Comment by lbroudoux** [comment]
Can you check that the status of the CR now holds references to the discovered API names and versions? This is the way to check you are running the correct version of the operator.

**Comment by mmatur** [comment]
I have done a new test with a fresh install with news CRDs version

And still have the issue, the operator is raising errors

<img width="899" alt="image" src="https://github.com/user-attachments/assets/fd1b5325-4e9e-474a-8deb-ba1e9b93577e">

Is there a way to enable more verbose logging?

**Comment by lbroudoux** [comment]
Have you also updated the `version` property of your `Microcks` CR to `nightly`?

**Comment by mmatur** [comment]
Yes also updated the Microcks `version` to `nightly` but nothing change.

Let's discuss about that tomorrow in live like discussed and we will update the issue accordingly.

**Comment by lbroudoux** [comment]
Now fixed via #54
